### PR TITLE
Feature/attachment and mic buttons

### LINKED ISSUE #361: Add Attachment Menu Button UI Component
Add an attachment button (paperclip/plus icon) to the left side of the message input field that opens a menu with options for sending images, sharing location, and creating polls. This is the foundational UI component required for all other attachment features.

### LINKED ISSUE #372: Add Tests for Attachment Menu Button UI Component
Add comprehensive unit and UI tests for the attachment menu button component to ensure proper functionality, user interactions, and edge cases are covered.

### PR DESCRIPTION
**Pull request by Atilla13**
### Summary
This PR implements the attachment menu button UI component and its comprehensive test suite. The button provides foundational UI for attachment features including image sharing, location sharing, and poll creation.

#### Implementation (#361)
- Added attachment button (paperclip/plus icon) to the left side of the message input field
- Implemented menu with options for:
  - Sending images
  - Sharing location
  - Creating polls
- Added Audio Recording Button with smart toggle behavior:
  - Displays when the message input field is empty
  - Automatically switches to send button when user starts typing
  - Returns to audio button when input is cleared

#### Testing (#372)
- Added unit tests for component rendering and state management
- Implemented UI tests covering:
  - User interactions (button clicks, menu open/close)
  - Menu option selection
  - Edge cases and error handling
- Achieved comprehensive test coverage for all component functionality

### Related Issues
- Closes #361
- Closes #372

### Testing Done
- ✅ Unit tests pass
- ✅ UI tests pass
- ✅ Manual testing on Android emulator
- ✅ Code formatting with ktfmt
- ✅ No new SonarCloud issues

### Demonstration

https://github.com/user-attachments/assets/0871bd39-c25f-4528-a474-a4fb8be4b417

### Notes
- This PR provides UI components only - the actual functionality for images, location sharing, polls, and audio recording will be fully implemented in future PRs
- Component follows existing chat UI patterns and design system
- All tests follow project testing conventions using JUnit and Compose testing libraries

---
_This PR description was written with assistance from Claude AI_

### CODE REVIEW

File: app/src/main/java/com/android/joinme/ui/chat/ChatScreen.kt [hunk 1]
```diff
@@ -447,13 +467,28 @@ private fun MessageInput(
...
         horizontalArrangement = Arrangement.spacedBy(Dimens.Spacing.small)) {
+          // Attachment button (left)
+          IconButton(
+              onClick = { showAttachmentMenu = true },
+              modifier =
+                  Modifier.size(Dimens.TouchTarget.minimum)
+                      .testTag(ChatScreenTestTags.ATTACHMENT_BUTTON)) {
+                Icon(
+                    imageVector = Icons.Default.AttachFile,
+                    contentDescription = "Add attachment",
```
Comment (MathieuPFR) [inline app/src/main/java/com/android/joinme/ui/chat/ChatScreen.kt]: I think, as the coaches told us multiples times it's a nice practice to use the resource files strings.xml for the hardcoded strings. This improves localization, consistency, and test stability. Like 
<*resources>
  <*string name="add_attachment">Add attachment</string>
  <*string name="record_audio">Record audio</string>
  <*string name="send_message">Send message</string>
  <*string name="gallery">Gallery</string>
  <*string name="location">Location</string>
  <*string name="poll">Poll</string>
<*/resources>
(I put * so that you can see in the message)
and then in your code : 
Icon(
  imageVector = Icons.Default.AttachFile,
  contentDescription = stringResource(R.string.add_attachment),
  tint = MaterialTheme.colorScheme.onSurfaceVariant
)
or Text(text = stringResource(R.string.gallery))

and then you can re use it in the composeTestRule.onNodeWithText

File: app/src/main/java/com/android/joinme/ui/chat/ChatScreen.kt [hunk 2]
```diff
@@ -467,27 +502,137 @@ private fun MessageInput(
...
-                    contentDescription = "Send message button",
-                    tint =
-                        if (text.isNotBlank()) onSendButtonColor
-                        else MaterialTheme.colorScheme.onSurfaceVariant)
-              }
+          // Dynamic send/mic button (right)
+          if (text.isEmpty()) {
+            // Microphone button (placeholder for future audio recording)
+            IconButton(
+                onClick = { /* TODO(#364 and #367): Audio recording - Feature coming soon */},
```
Comment (MathieuPFR) [inline app/src/main/java/com/android/joinme/ui/chat/ChatScreen.kt]: It's nice that you put a TODO comment for the feature implementation coming soon with the # issue but try to follow what we did until now : put a Toast("Not implemented yet") so that it is properly handled when using the app. Even if it's gonna be corrected soon it's better for the user experience.

**Comment by Atilla13** [comment]
Thank you @MathieuPFR for taking the time to review my PR! Your feedback is really helpful, and I’ll make sure to update everything you pointed out.
